lale/operators.py

```diff
@@ -3850,11 +3859,7 @@ def __init__(
                         ]
                     )
                     new_edges.extend(
-                        [
-                            edge
-                            for edge in edges
-                            if (edge[1] != tstep and edge[0] != tstep)
-                        ]
+                        [edge for edge in edges if tstep not in (edge[0], edge[1])]
```

Comment:
Can omit the square brackets: no need to materialize the argument as a list, just pass the generator directly.